examples/sycl/pvc/flash_attention_v2/pvc_flash_attn_gemm_universal.hpp

```diff
@@ -382,69 +356,54 @@ class GemmUniversalAttention
       collective_mma.mmaQK(tile_coord_QK, tSr, gQ, gK, tSr, head_size / get<1>(subgroup_shape), params.mainloop);
 
       // Apply causal mask
-      if constexpr (CausalMask)
-      {
+      if (CausalMask && nblock == nblock_limit - 1) {
         // mask the elements of each tile where j > i
         int col_idx = item_id + load_idx;
         CUTLASS_PRAGMA_UNROLL
-        for (int n = 0; n < FragsN; n++, col_idx += get<1>(MmaAtomShape()))
-        {
+        for (int n = 0; n < FragsN; n++, col_idx += get<1>(MmaAtomShape())) {
           CUTLASS_PRAGMA_UNROLL
-          for (int m = 0; m < FragsM; m++)
-          {
+          for (int m = 0; m < FragsM; m++) {
             int row_idx = m * Vec + seq_coord;
             CUTLASS_PRAGMA_UNROLL
-            for (int row = 0; row < Vec; row++, row_idx++)
-            {
+            for (int row = 0; row < Vec; row++, row_idx++) {
               if (col_idx > row_idx)
                 tSr(row, m, n) = -INFINITY;
             }
           }
         }
       }
-      
-      flash::Softmax<ElementAccumulator>::template run<CausalMask, Vec, FragsM, FragsN>(nblock == 0, tSr,
-                                                                                        max_reg, sum_reg, out_reg, params.softmax);
+
+      flash::Softmax<ElementAccumulator>::template run<CausalMask, Vec, FragsM, FragsN>(
+          nblock == 0, tSr, max_reg, sum_reg, out_reg, params.softmax);
       // 7) Convert S to P (FP32 -> BF16)
       Tensor tPr = make_tensor<typename TiledMma::ValTypeA>(shape(tSr));
       CUTLASS_PRAGMA_UNROLL
-      for (int p_idx = 0; p_idx < size(tPr); p_idx++)
-      {
-        #ifdef __SYCL_DEVICE_ONLY__
+      for (int p_idx = 0; p_idx < size(tPr); p_idx++) {
+#ifdef __SYCL_DEVICE_ONLY__
         // Temporary patch to avoid linking in the devicelib fallback unconditionally.
         tPr(p_idx).storage = __spirv_ConvertFToBF16INTEL(tSr(p_idx));
```

Comment:
For AOT this has been fixed: https://github.com/intel/llvm/pull/16494 . Do we want to keep the workaround?

Reply:
Thanks For pointing it out. We are trying to address that properly in a separate PR : https://github.com/codeplaysoftware/cutlass-fork/pull/203/files. This is useful to know that it has been fixe, so the #PR203 can apply it and fix it